src/sentry/models/lostpasswordhash.py

```diff
@@ -67,17 +73,8 @@ def send_email(self, request, mode="recover"):
         msg.send_async([self.user.email])
 
     @classmethod
-    def for_user(cls, user):
-        # NOTE(mattrobenolt): Some security people suggest we invalidate
-        # existing password hashes, but this opens up the possibility
-        # of a DoS vector where then password resets are continually
-        # requested, thus preventing someone from actually resetting
-        # their password.
-        # See: https://github.com/getsentry/sentry/pull/17299
-        password_hash, created = cls.objects.get_or_create(user=user)
-        if not password_hash.is_valid():
-            password_hash.date_added = timezone.now()
-            password_hash.set_hash()
-            password_hash.save()
+    def for_user(cls, user) -> "APILostPasswordHash":
+        from sentry.services.hybrid_cloud.lostpasswordhash import lost_password_hash_service
 
+        password_hash = lost_password_hash_service.get_or_create(user.id)
```

Comment:
If LostPasswords are control only what does the service do?

Reply:
It essentially updates attributes on the user

Reply:
Should we roll that into the users service then?

Reply:
Er, that last comment was not very accurate. LostPasswordHash is a separate model that is related to a user. It's super lightweight though. Only has the hash and a date. I could go either way on combining it with user service.